main/openrc/use-tod-as-clocksource-for-s390x.patch

```diff
@@ -0,0 +1,37 @@
+diff --git a/init.d/hwclock.in b/init.d/hwclock.in
+index 024bff0..bfe501c 100644
+--- a/init.d/hwclock.in
++++ b/init.d/hwclock.in
+@@ -41,7 +41,7 @@ setupopts()
+ {
+ 	case "$(uname -m)" in
+ 		s390*)
+-			utc="s390"
++			utc="UTC"
+ 			;;
+ 		*)
+ 			if [ -e /proc/devices ] && \
+@@ -118,10 +118,12 @@ start()
+ 		: $(( retval += $? ))
+ 	fi
+ 
+-	if yesno ${clock_hctosys:-YES}; then
++	if [ "$(uname -m)" = "s390x" ]; then
++		date -s "@$(gettod)"
++	elif yesno ${clock_hctosys:-YES}; then
+ 		_hwclock --hctosys $utc_cmd $(get_noadjfile) $clock_args
+-		: $(( retval += $? ))
+ 	fi
++	: $(( retval += $? ))
+ 
+ 	eend $retval "Failed to set the system clock"
+ 
+@@ -131,7 +133,7 @@ start()
+ stop()
+ {
+ 	# Don't tweak the hardware clock on LiveCD halt.
+-	[ -n "$CDBOOT" ] && return 0
++	[ -n "$CDBOOT" ] || [ "$(uname -m)" = "s390x" ] && return 0
```

Comment:
Can we change this to an `if ...; then ...` statement instead? Its more readable.